### PR TITLE
Improve scan plugins

### PR DESCRIPTION
**Pull request by Fyzu**

### CODE REVIEW

**Comment by Fyzu** [comment]
[61 Improve plugin scan in core](https://trello.com/c/mGs24dhb/61-61-improve-plugin-scan-in-core)

File: scanActivators.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  const lernaJsonPath = path.resolve(projectPath, LERNA_JSON)
+  const { packages: lernaPackages } = fs.readJsonSync(lernaJsonPath, { throws: false }) || {}
+  if (!lernaPackages) {
+    throw new Error('Incorrect configuration file `lerna.json`')
+  }
+
+  const lernaPluginsPaths = lernaPackages.map(pluginsPath => path.resolve(projectPath, `./${pluginsPath}`))
+  const pluginsPaths = lernaPluginsPaths
+    .map(pluginsPath => path.resolve(pluginsPath, NODE_MODULES_PLUGINS_PATH))
+    .concat(path.resolve(projectPath, NODE_MODULES_PLUGINS_PATH))
```
Comment (wuzyk) [inline scanActivators.js]: In what case do we need to search activators in the `node_modules` of the project?

File: __mocks__/scanActivators.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+let mockActivators = []
+
+export const setMockActivators = activators => {
+  mockActivators = activators
+}
+
+export default function scanActivators() {
+  return mockActivators || []
+}
```
Comment (wuzyk) [inline __mocks__/scanActivators.js]: Such mock seems to be verbose. You can mock directly in your test like this:
```
jest.mock('../scanActivators')
const { scanActivators } = require.requireMock('../scanActivators')
...
scanActivators.mockImplementation(() => activators)
```

File: scanActivators.js (hunk 1 above)
Comment (Fyzu) [inline scanActivators.js]: For example, we want to add `@rispa/webpack` from npm, after installation it will be in the `node_modules` of the project.

Comment (wuzyk) [inline scanActivators.js]: Only if we add it as a dependency to package.json of the project, but not by using `ris add`, right?
But why do we need to add plugin such way?

Comment (Fyzu) [inline scanActivators.js]: Right.
In order not to store non-extendable plugins in `packages/*`.

Comment (wuzyk) [inline scanActivators.js]: ok, got it

**Pull request merged**